src/node_util.cc

```diff
@@ -64,13 +64,15 @@ static void GetPromiseDetails(const FunctionCallbackInfo<Value>& args) {
   auto isolate = args.GetIsolate();
 
   Local<Promise> promise = args[0].As<Promise>();
-  Local<Array> ret = Array::New(isolate, 2);
 
   int state = promise->State();
-  ret->Set(env->context(), 0, Integer::New(isolate, state)).FromJust();
+  std::vector<Local<Value>> values;
```

Comment:
`values(2)`, as we can have a maximum of two values.

Reply:
Then way not simply:
```suggestion
  Local<Value> values[2] = { Integer::New(isolate, state) };
  size_t number_of_values = 1;
  if (state != Promise::PromiseState::kPending)
    values[number_of_values++] = promise->Result();
  DCHECK_LT(number_of_values, arraysize(values));
  Local<Array> ret = Array::New(isolate, values, number_of_values);
```

Reply:
That’s much longer and doesn’t really provide too much of an advantage over using a vector.

Reply:
AFAICT it's only one line longer, and we get everything on the stack with no free-store allocations.
I thought that when possible, we prefer static sized data structures?

Reply:
I tried @refack's suggestion, but I couldn't find a reasonable way to include DCHECK_LE in this file. DCHECK_LE is defined in `deps/v8/src/base/logging.h`, and when I add `#include "../deps/v8/src/base/logging.h"` at the top, the compile doesn't seem passing (relative path inside logging.h doesn't seem resolving...)

Reply:
I followed @TimothyGu's comment for the moment.

Reply:
With `values(2)`, the values are initialized with empty values and it has size 2. So I needed to add an `else` block and `.pop_back()` the last (empty) value. (I think that's unnecessarily complex and not desirable code.)

So I changed the code to follow @refack's comment (without DCHECK_LE assertion).